docs/contributing/contributing-code.md

```diff
@@ -230,3 +230,42 @@ On M1 macs some `pre-commit` tests may fail.
 If those checks are running in docker containers (such as `hadolint`) and exit with code 137, this means they are running into a memory limit.
 This is because for running x86_64 images on an M1 mac, docker will utilize an emulation with qemu that can cause higher memory consumption.
 To fix this, change the memory limit in Docker Desktop in the Resources section of the Preferences, 8.00GB should be sufficient.
+
+### Working with git submodules
+
+A submodule is a reference to a specific commit in another git
+repository. This results in a subfolder containing a checkout of that
+repository at that particular commit. Submodules have a reputation of
+being tricky to use. To minimize problems, make sure to follow these
+guidelines:
+
+* When checking out a new branch or commit, update the submodules
+  using the command `git submodule update --init --recursive`.
+  Adding a shortcut to your shell can be useful. The following is a
+  Bash function that lets you call `gitup`. It goes into your `~/.bashrc`:
+
+```bash
+gitup() {
+  echo "git submodule update --init --recursive"
+  git submodule update --init --recursive
+}
+```
+
+* When modifying both a parent repo A and one of its submodules B,
+  make one pull request for each (PR A, PR B).
+  1. Before merging PR B, make sure the branch on repo B is **not
+     lagging behind** the main branch. This ensures that the submodule
+     includes all the latest changes made by others.
+  2. Make sure PR B is merged **before** PR A.
+     This ensures that other developers will pick up the changes on B
+     when making their own changes.
+  3. After merging PR A, check that submodule B is still up-to-date
+     with respect to its main branch, especially if PR B was merged
+     more than an hour ago.
+  Good to know:
+  - Merging in B can be done with a merge commit or by squashing the
+    commits.
+  - If squashing commits in B, the original commit referenced by A becomes
```

Comment:
Oh it's mentioned here. I personally don't agree with this practice. It may not cause any immediate problems, but I also care about being able to build things in the future.

Reply:
I agree but I don't want to spend more time on this. I want people to understand the consequences of their actions (squashing commits and not pointing to the squashed commit). For some people, it can speed things up a bit because the submodule gets updated frequently. I'll let them decide what's best and hold them responsible for any ensuing success or failure.

Reply:
I added a warning at the end of the paragraph.